library/core/src/ptr/non_null.rs

```diff
@@ -762,6 +768,14 @@ impl<T: ?Sized> NonNull<T> {
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
     #[stable(feature = "non_null_convenience", since = "1.80.0")]
     #[rustc_const_stable(feature = "non_null_convenience", since = "1.80.0")]
+    #[kani::should_panic]
+    #[requires(
+        (self.as_ptr() as usize == origin.as_ptr() as usize) || 
+        (kani::mem::same_allocation(self.as_ptr(), origin.as_ptr()) && 
+         ((self.as_ptr() as usize >= origin.as_ptr() as usize) && 
```

Comment:
I don't think the function has this requirement. The result can be positive (if `origin` is less than or equal to `self`) and negative otherwise.